### PR TITLE
C namespacing

### LINKED ISSUE #9: Duplicate symbol blake2s
When using `haskell-language-server` to check a Haskell module in my project which transitively depends on both `crypton` and `cryptonite`, I received the following error. (This detail is shown when `-d` (debug) flag is turned on in `haskell-language-server`). Howerver, the project works fine with both `cabal build` and `cabal repl`.

By looking at the source code of `crypton`, I noticed that in `cbits/blake2/sse/blake2s.c` file, there is a non-static function named `blake2s`. This is a possible cause of symbol conflicts.

```text
GHC runtime linker: fatal error: I found a duplicate definition for symbol
   blake2s
whilst processing object file
   /nix/store/pjy8nwfnn2cpzfbwlzc5ib7vj05w92pl-crypton-lib-crypton-0.32/lib/x86_64-linux-ghc-9.4.5/crypton-0.32-EGpwvAbNtQg1OZ3fW00GIA/HScrypton-0.32-EGpwvAbNtQg1OZ3fW00GIA.o
The symbol was previously defined in
   /nix/store/jn06hq2n07xwd2zalfjmnc8cwri3xxxf-cryptonite-lib-cryptonite-0.30/lib/x86_64-linux-ghc-9.4.5/cryptonite-0.30-5urQnFo6cSD563Pl2BMv1A/HScryptonite-0.30-5urQnFo6cSD563Pl2BMv1A.o
This could be caused by:
   * Loading two different object files which export the same symbol
   * Specifying the same object file twice on the GHCi command line
   * An incorrect `package.conf' entry, causing some object to be
     loaded twice.
```

Suggested fix: rename all non-static C symbols in `crypton` to be prefixed with `crypton_`.

### PR DESCRIPTION
**Pull request by endgame**
Fixes #9

### CODE REVIEW

**Comment by endgame** [comment]
Pretty sure that's the last of them, because I ran this before I started:
```
[.../ghc-9.0.2/crypton-0.32/build]$ nm -D libHScrypton-0.32-inplace-ghc9.0.2.so | grep -v crypton | grep -v ghc | grep -v '^ *U'
0000000000577430 T blake2s
00000000006209e8 D __bss_start
                 w __cxa_finalize@GLIBC_2.2.5
00000000006209e8 D _edata
0000000000620fa2 D _end
0000000000620810 D gfmul_branch_ptr
                 w __gmon_start__
                 w _ITM_deregisterTMCloneTable
                 w _ITM_registerTMCloneTable
```

File: cbits/blake2/sse/blake2s.c [hunk 1]
```diff
@@ -290,7 +290,7 @@ int blake2s( void *out, size_t outlen, const void *in, size_t inlen, const void
 #if defined(SUPERCOP)
 int crypto_hash( unsigned char *out, unsigned char *in, unsigned long long inlen )
```
Comment (kokobd) [inline cbits/blake2/sse/blake2s.c:291]: Is this inteded to be `crypto_` or is a typo?

Comment (endgame) [reply cbits/blake2/sse/blake2s.c:291]: It's that way in the [official source](https://github.com/BLAKE2/BLAKE2/blob/master/sse/blake2s.c#L291), so I assume it's designed to interface with the [`SUPERCOP` benchmark suite](https://bench.cr.yp.to/supercop.html).

**Comment by kokobd** [comment]
I confirm `haskell-language-server` works now.

**Review by kazu-yamamoto — APPROVED** [review]
LGTM

**Comment by kazu-yamamoto** [comment]
Merged and released.
Thank you for your contribution!

**Comment by endgame** [comment]
Thanks for the swift action.

**Pull request merged**